### PR TITLE
Upgrade Eslint rules - August-September, 2021

### PR DESCRIPTION
**Pull request by JustBeard**
- **[deps]** update: `eslint@7.32.0`, `eslint-config-airbnb-base@14.2.1`, `eslint-config-prettier@8.3.0`, `eslint-plugin-import@2.24.2`, `eslint-plugin-prettier@4.0.0`, `prettier@2.4.0`
- **[deps]** update instructions and support: `eslint-config-airbnb@18.2.1`, `eslint-plugin-jsx-a11y@6.4.1`, `eslint-plugin-react@7.25.1`, `eslint-plugin-react-hooks@4.2.0`

- **[config]** eslint - ecmaVersion: 2021
- **[config]** eslint - enable react hooks
- **[config]** prettier - define `"arrowParens": "avoid"` as default value changed
- **[config]** prettier - new config `"embeddedLanguageFormatting": "auto"`

- **[new rule]** default-case-last
- **[new rule]** no-loss-of-precision
- **[new rule]** no-promise-executor-return
- **[new rule]** no-unreachable-loop
- **[new rule]** no-unsafe-optional-chaining
- **[new rule]** import/no-relative-packages
- **[new rule]** import/no-import-module-exports
- **[new rule]** react/sort-prop-types
- **[new rule]** react/jsx-sort-props
- **[new rule]** react/no-unstable-nested-components
- **[new rule]** react/jsx-no-constructed-context-values
- **[new rule]** react/function-component-definition

- **[new option]** react/jsx-no-target-blank: { forms: true, warnOnSpreadAttributes: true }
- **[new option]** react/jsx-filename-extension: { allow: 'as-needed' }

- **[configuration]** react/sort-comp sorting extended with new pattern '/^handle.+$/'

- **[chore]** initiate CHANGELOG

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# 3.0.0 / 2021-09
```
Comment (y-nk) [inline CHANGELOG.md:1]: imho should be automated (see #12)

File: README.md [hunk 2]
```diff
@@ -1,42 +1,76 @@
...
-- `eko` - base js rules, import rules
-- `eko/react` - `eko`, react rules, jsx-a11y rules
+- `eko` - eslint, import rules
+- `eko/react` - `eko` + react, react-hook, jsx-a11y rules
 
-## Pre-installation:
+## Installation:
 
-Copy `.prettierrc` file into root of your project. Make sure that you do not change configuration or you will run in a bunch of conflicts between prettier and eslint.
+### Step 1. Mandatory. Copy this configuration to your `.prettierrc` file:
```
Comment (y-nk) [inline README.md:12]: I think we can add a `.prettierrc` file in here and use this to have inheritance: https://github.com/prettier/prettier/pull/5963

**Review by y-nk — APPROVED** [review]
lgtm, made few optional comments

**Review by nuipawarit — APPROVED** [review]
LGTM

**Review by beermindnor — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (JustBeard) [reply README.md:12]: @y-nk 
Not really keen to publish a new package just for few lines. But I think that we can extend as
```
"prettier": "eslint-config-eko/prettier.json"
```
It is something I can test more.

On other hand, `Prettier` stated that they would not be adding or changing configurations anymore. Considering 

There is as well an issue. We have few projects where I am not sure how the extension will work. Consider structure below:
```
big_adonis_codebase
  |- ...
  |- react
    |- new plugged in React components (Adonis to React migration)
    |- eslint and prettier configuration
  |- ...
  package.json - with scripts to run adonis and react
```
In the current configuration (`prettier: "external_conf"`) at the root level might cause an unwanted effect on Adonis codebase. Not sure now how to handle it.

**Review by frankent — APPROVED** [review]

**Pull request merged**